src/app/counter-analytics/day-of-week-histogram/day-of-week-histogram.page.component.ts

```diff
@@ -0,0 +1,46 @@
+import { AfterViewInit, Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { CounterAnalyticsService } from '../../services/counter-analytics.service';
+import 'anychart';
+import { Counter } from '../../models/counter';
+
+@Component({
+  selector: 'app-counter-analytics',
+  templateUrl: './day-of-week-histogram.page.component.html',
+  styleUrls: ['./day-of-week-histogram.page.component.scss'],
+})
+export class DayOfWeekHistogramPage implements OnInit, AfterViewInit {
+
+  counter: Counter;
+  chart: anychart.charts.Cartesian = anychart.column();
+
+  @ViewChild('chartContainer', {static: true}) container: ElementRef;
+
+  constructor(
+    private route: ActivatedRoute,
+    private counterAnalytics: CounterAnalyticsService
+  ) {}
+
+  ngOnInit() {
+    const weekDays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    this.counter = this.route.snapshot.data.counter;
+
+    const positiveData = this.counterAnalytics.getDayOfWeekHistogramData(this.counter, 'positive');
+    const positiveSeries = this.chart.column(positiveData.map((value, index) => [weekDays[index], value]));
+    positiveSeries.name('Plus Count');
+    positiveSeries.stroke('green');
+    positiveSeries.fill('green');
+
+    const negativeData = this.counterAnalytics.getDayOfWeekHistogramData(this.counter, 'negative');
+    const negativeSeries = this.chart.column(negativeData.map((value, index) => [weekDays[index], -value]));
+    negativeSeries.name('Minus Count');
+    negativeSeries.stroke('red');
+    negativeSeries.fill('red');
+  }
+
+  ngAfterViewInit() {
+    this.chart.container(this.container.nativeElement);
+    this.chart.draw();
```

Comment:
* x axis- should always display all days maybe slanted slightly so that they all fit
* y axis should auto scale but only display natural numbers
* we should add a label telling the user what the bar color stands for, maybe?

Reply:
Agree. For the x axis I was also thinking about defining a minimum size and if it does not fit the user can scroll horizontally.

Reply:
Implemented all of your points.